### PR TITLE
jewel: tests: New upgrade test for #19508

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/19829

Implements the upgrade test discussed in #14392

### CODE REVIEW

**Comment by smithfarm** [comment]
@asheplyakov @jdurgin Initial attempt to implement the upgrade test discussed in #14392

**Comment by smithfarm** [comment]
Note: #14626 is related to this!

File: qa/suites/upgrade/hammer-x/v0-94-6-mon-overload/v0-94-6-mon-overload.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+- install:
+    tag: v0.94.6
+- ceph:
+    fs: xfs
+- print: "*** Upgrade the target that runs osd.0 and osd.1 to -x while the target that runs osd.2 stays v0.94.6"
+- install.upgrade:
+    osd.0:
+- print: "*** Restart mon.a and osd.0 so they run jewel
+- ceph.restart:
+    daemons: [mon.a, osd.0, osd.1]
```
Comment (asheplyakov) [inline qa/suites/upgrade/hammer-x/v0-94-6-mon-overload/v0-94-6-mon-overload.yaml]: I think only osd.0 should be restarted here (so only osd.0 runs Jewel)

Comment (smithfarm) [inline qa/suites/upgrade/hammer-x/v0-94-6-mon-overload/v0-94-6-mon-overload.yaml]: The idea is to restart osd.0 and osd.1 so they run jewel, and leave osd.2 (on a separate machine) running hammer. Updated the preceding print to clarify.

**Comment by smithfarm** [comment]
Jenkins re-test this please

**Comment by smithfarm** [comment]
`./virtualenv/bin/teuthology-suite -k distro --verbose --suite upgrade/hammer-x/v0-94-6-mon-overload --ceph-repo https://github.com/ceph/ceph --ceph jewel --suite-repo https://github.com/smithfarm/ceph --suite-branch wip-19829-jewel --machine-type vps --priority 101 --email ncutler@suse.com`

**waiting for Jenkins clock skew issue to be addressed**

**Comment by smithfarm** [comment]
Jenkins re-test this please

**Comment by smithfarm** [comment]
Changelog:

* rebased to latest jewel
* added `ceph osd set sortbitwise` after rebooting last mon to jewel
* fixed an awful syntax error!

**Comment by smithfarm** [comment]
`teuthology-suite -k distro --verbose --suite upgrade/hammer-x/v0-94-6-mon-overload --ceph-repo https://github.com/ceph/ceph --ceph jewel --suite-repo https://github.com/smithfarm/ceph --suite-branch wip-19829-jewel --machine-type vps --priority 101 --email ncutler@suse.com --teuthology-branch wip-20392`

**running** http://pulpito.front.sepia.ceph.com:80/smithfarm-2017-06-27_00:12:56-upgrade:hammer-x:v0-94-6-mon-overload-jewel-distro-basic-vps/

**Comment by smithfarm** [comment]
**running** http://pulpito.front.sepia.ceph.com:80/smithfarm-2017-06-27_01:31:00-upgrade:hammer-x:v0-94-6-mon-overload-jewel-distro-basic-vps/

**Comment by smithfarm** [comment]
Running with `--num 5`

**running** http://pulpito.front.sepia.ceph.com:80/smithfarm-2017-06-27_01:51:29-upgrade:hammer-x:v0-94-6-mon-overload-jewel-distro-basic-vps/

**Comment by smithfarm** [comment]
**running** http://pulpito.front.sepia.ceph.com:80/smithfarm-2017-06-27_07:23:34-upgrade:hammer-x:v0-94-6-mon-overload-jewel-distro-basic-vps/

**Comment by smithfarm** [comment]
Testing against the fix from #14392 - should succeed now.

**running** http://pulpito.front.sepia.ceph.com:80/smithfarm-2017-06-27_10:05:31-upgrade:hammer-x:v0-94-6-mon-overload-wip-jewel-backports-distro-basic-vps/

**Comment by smithfarm** [comment]
@asheplyakov You were right; I needed to restart only `osd.0`. Fixed now.

The test fails against jewel and succeeds against `wip-jewel-backports` (which contains #14392)

**Comment by smithfarm** [comment]
@jdurgin I completely reworked this PR - could you take another look? It fails on baseline jewel (for the right reason) and passes on jewel + #14392.

**Pull request merged**